harness/determined/estimator/_estimator_trial.py

```diff
@@ -49,6 +49,23 @@
 VERY_LARGE_NUMBER = 9999999999999999
 
 
+class DeterminedEarlyStoppingHook(tf.compat.v1.train.SessionRunHook):  # type: ignore
```

Comment:
non-blocking: we should add a unit test for this. Something along the lines of having a callback that stops training after and evaluation and then use our checkpointing callback to make sure that everything executes as expected.